src/barril/units/_scalar.py

```diff
@@ -313,8 +316,19 @@ def _DoOperation(self, p1, p2, operation, callback_operation):
             )
 
         unit_database = self._unit_database
-        operation = getattr(unit_database, operation)
-        q, v = operation(p1.GetQuantity(), p2.GetQuantity(), self._value, p2.value)
+        operation_func = getattr(unit_database, operation)
+        if p1_is_number:
+            assert operation in [
+                "Divide",
+                "FloorDivide",
+            ], "Only operation Divide and FloorDivide allowed here!"
+            q, v = operation_func(
+                p2.GetQuantity(), p2.GetQuantity() * p2.GetQuantity(), p1, p2.value
```

Comment:
I am a bit curious.
Why does `p2.GetQuantity(), p2.GetQuantity() * p2.GetQuantity(), p1, p2.value` works?

Reply:
Now that you've questioned this approach, I've changed to something more clear I think:

```python
q, v = operation_func(
   Quantity.CreateEmpty(), p2.GetQuantity(), p1, p2.value
)
```

Reply:
Ah....
Nice =)